aten/src/ATen/native/native_functions.yaml

```diff
@@ -8937,4 +8937,4 @@
 - func: segment_reduce(Tensor data, str reduce, *, Tensor? lengths=None, Tensor? indices=None, int axis=0, bool unsafe=False) -> Tensor
```

Comment:
as we've discussed, add `initial` argument here to handle 0-length case

Reply:
will leave this change to another PR on the stack in favor of smaller PRs.